src/Sylius/Bundle/ApiBundle/Validator/Constraints/PromotionCouponEligibilityValidator.php

```diff
@@ -43,29 +47,44 @@ public function validate(mixed $value, Constraint $constraint): void
             return;
         }
 
+        /** @var PromotionCouponInterface|null $promotionCoupon */
```

Comment:
I think this docblock shouldn't be necessary since it's configured at constructor.